### PR TITLE
disable shipping link for all virtual types, DeliveryTime and Weight too

### PR DESCRIPTION
**Pull request by Hanmac**
- This disables the Shopping Link for Virtual Types
- Also DeliveryTime is skipped for Virtual ones. (or otherwise need to be better configured?)
- Also Weight Info is skipped, i think the weight calculation in Core Magento skips Virtual Products anyway.

### CODE REVIEW

File: src/app/code/community/FireGento/MageSetup/Block/Bundle/Catalog/Product/Price.php [hunk 1]
```diff
@@ -116,13 +116,7 @@ public function isIncludingTax()
...
     public function isShowShippingLink()
     {
-        $productTypeId = $this->getProduct()->getTypeId();
-        $ignoreTypeIds = array('virtual', 'downloadable');
-        if (in_array($productTypeId, $ignoreTypeIds)) {
-            return false;
-        }
-
-        return true;
+        return !$this->getProduct()->isVirtual();
```
Comment (sprankhub) [inline src/app/code/community/FireGento/MageSetup/Block/Bundle/Catalog/Product/Price.php:119]: I thought this check is missing downloadable products, but actually, it includes it plus, it includes more things like bundle products with only virtual children. Nice!

File: src/app/code/community/FireGento/MageSetup/Block/Bundle/Catalog/Product/Price.php [hunk 2]
```diff
@@ -156,6 +150,9 @@ protected function _loadTaxCalculationRate(Mage_Catalog_Model_Product $product)
      */
     public function getIsShowWeightInfo()
     {
+        if ($this->getProduct()->isVirtual()) {
+            return;
```
Comment (sprankhub) [inline src/app/code/community/FireGento/MageSetup/Block/Bundle/Catalog/Product/Price.php]:
```suggestion
            return false;
```

File: src/app/code/community/FireGento/MageSetup/Block/Bundle/Catalog/Product/Price.php [hunk 3]
```diff
@@ -166,6 +163,9 @@ public function getIsShowWeightInfo()
      */
     public function getFormattedWeight()
     {
+        if ($this->getProduct()->isVirtual()) {
+            return "";
```
Comment (sprankhub) [inline src/app/code/community/FireGento/MageSetup/Block/Bundle/Catalog/Product/Price.php]:
```suggestion
            return '';
```

File: src/app/code/community/FireGento/MageSetup/Block/Bundle/Catalog/Product/Price.php [hunk 4]
```diff
@@ -166,6 +163,9 @@ public function getIsShowWeightInfo()
      */
     public function getFormattedWeight()
     {
+        if ($this->getProduct()->isVirtual()) {
```
Comment (sprankhub) [inline src/app/code/community/FireGento/MageSetup/Block/Bundle/Catalog/Product/Price.php]:
```suggestion
        if (!$this->getIsShowWeightInfo()) {
```

Wouldn't that even make more sense?

File: src/app/code/community/FireGento/MageSetup/Block/Catalog/Product/Price.php [hunk 5]
```diff
@@ -258,6 +258,9 @@ public function getIsShowWeightInfo()
      */
     public function getFormattedWeight()
     {
+        if ($this->getProduct()->isVirtual()) {
+            return "";
```
Comment (sprankhub) [inline src/app/code/community/FireGento/MageSetup/Block/Catalog/Product/Price.php]:
```suggestion
            return '';
```

File: src/app/code/community/FireGento/MageSetup/Block/Catalog/Product/Price.php [hunk 6]
```diff
@@ -258,6 +258,9 @@ public function getIsShowWeightInfo()
      */
     public function getFormattedWeight()
     {
+        if ($this->getProduct()->isVirtual()) {
```
Comment (sprankhub) [inline src/app/code/community/FireGento/MageSetup/Block/Catalog/Product/Price.php]:
```suggestion
        if (!$this->getIsShowWeightInfo()) {
```

Wouldn't that even make more sense?

File: src/app/code/community/FireGento/MageSetup/Block/Bundle/Catalog/Product/Price.php (hunk 4 above)
Comment (Hanmac) [reply src/app/code/community/FireGento/MageSetup/Block/Bundle/Catalog/Product/Price.php]: you need to negate the if condition there

File: src/app/code/community/FireGento/MageSetup/Block/Catalog/Product/Price.php (hunk 6 above)
Comment (Hanmac) [reply src/app/code/community/FireGento/MageSetup/Block/Catalog/Product/Price.php]: you need to negate the if condition there

**Comment by Hanmac** [comment]
for Bundle stuff, it would need more JS changes, because it might not work correctly when a Bundle was virtual before, but with added Items it isn't virtual anymore. current Magento JS can't handle such change, but that isn't probably the goal of this MR

File: src/app/code/community/FireGento/MageSetup/Block/Bundle/Catalog/Product/Price.php (hunk 4 above)
Comment (sprankhub) [reply src/app/code/community/FireGento/MageSetup/Block/Bundle/Catalog/Product/Price.php]: True :see_no_evil: Fixed!

File: src/app/code/community/FireGento/MageSetup/Block/Catalog/Product/Price.php (hunk 6 above)
Comment (sprankhub) [reply src/app/code/community/FireGento/MageSetup/Block/Catalog/Product/Price.php]: True :see_no_evil: Fixed!

**Comment by sprankhub** [comment]
> for Bundle stuff, it would need more JS changes, because it might not work correctly when a Bundle was virtual before, but with added Items it isn't virtual anymore. current Magento JS can't handle such change, but that isn't probably the goal of this MR

For dynamic changes, yes. But the bundle product type model also does some interesting virtual checks by default :)

**Review by sprankhub — APPROVED** [review]

**Pull request merged**